api/envoy/config/filter/network/rbac/v2/rbac.proto

```diff
@@ -27,4 +27,21 @@ message RBAC {
 
   // The prefix to use when emitting statistics.
   string stat_prefix = 3 [(validate.rules).string.min_bytes = 1];
+
+  enum EnforcementType {
+    // Apply RBAC policies during connection establishment to upstream.
+    ON_CONNECT = 0;
```

Comment:
Is this really happening during connection establishment as the comment stated? From the code it's actually happening on the first call of `onData()` which seems it's after the connection established, should the comment be updated to reflect this?

Reply:
From an enduser perspective, its the same isn't it? WE dont talk about our internal implementation to end users.. To them, they need to know if this is a one-time event done when establishing connection to upstream or if its going to be continuously checked..

Reply:
I agree with @yangminzhu, it does matter, since there might be significant delay between connection establishment and the first bit of data. I would rename ON_INITIAL_DATA or something like that, and leave us room to add ON_CONNECT later if needed (and also clarify docs).

Reply:
Okay updated to ONE_TIME_ON_FIRST_BYTE